### ISSUE TITLE
Introduce a way to block PRs when the last daily test failed

### ISSUE THREAD

**Issue by madolson** [madolson/placeholderkv#1465]
I think it would be useful to introduce some forcing function for us as a team to make sure the daily tests are staying green. My suggestion is we implement a check on the CI that looks into the last daily run and only allows code to get merged if it was green. @valkey-io/valkey-committers thoughts?

**Comment by ranshid** [comment]
I think it is a good idea which will help motivate us to keep the daily green. 
Why part of the ci though? maybe it is better to create another status check workflow which will use the github API to check the last daily status?

**Comment by hwware** [comment]
I agree with this, but I think we should tag the PR that we can merge but not merge due to CI issue.  such as "To be merged" ?

**Comment by hpatro** [comment]
I guess we would need something like this https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue. Otherwise, it would be quite painful for the group to figure out when the daily tests are green/stable and figure out list of PR which can be merged.

**Comment by madolson** [comment]
I'm not convinced a merge queue would solve this specific issue, but it's still something we've talked about adding quite often.

**Comment by hpatro** [comment]
> I'm not convinced a merge queue would solve this specific issue, but it's still something we've talked about adding quite often.

I think it will act as a forcing function when most of the PR would start getting stuck in the queue.

**Comment by hpatro** [comment]
@madolson Do you have the permission to enable the merge queue functionality ?

**Comment by madolson** [comment]
Yes, I can give you permissions as well if you want to set it up.

**Comment by hpatro** [comment]
> Yes, I can give you permissions as well if you want to set it up.

Sure, go ahead. I can try to set it up next week.

**Comment by madolson** [comment]
@hpatro You should have access to make changes now.
